utils/answer-types.js

```diff
@@ -1877,7 +1877,7 @@ Khan.answerTypes = $.extend(Khan.answerTypes, {
                     return {
                         empty: pass === "",
                         correct: pass === true,
-                        message: typeof pass === "string" ? pass : null,
+                        message: typeof pass === "string" && pass !== "" ? pass : null,
```

Comment:
Consider the simplest case of a custom-in-multiple (there actually are a few exercises with just this structure):

```html
<div class="solution" data-type="multiple">
    <div class="sol" data-type="custom">
        <div class="instruction">Always unanswered</div>
        <div class="guess">[2]</div>
        <div class="validator-function">return '';</div>
    </div>
</div>
```
(here's a minimum working exercise: [test-custom-in-multiple.html](https://github.com/wrwrwr/khan-exercises/blob/test-exercises/exercises/test-custom-in-multiple.html)). As the validator function always returns an empty string the problem should be considered unanswered.

The current custom exercise code first checks that the result is not an object and then returns a score that looks like this: ```{ empty=true,  correct=false,  guess=[1],  message=""}```. This becomes a "pass" in the multiple validator. The ```pass.message && pass.empty``` condition evaluates to an empty string and fails, thus the score (with its empty string message, with or without `empty` set to false) becomes the exercise score.

Further, the score is processed in `interface.js`. The message is not null, so at [line 477](https://github.com/Khan/khan-exercises/blob/master/interface.js#L477) `attemptMessage` is set to an empty string. However, the test at [line 484](https://github.com/Khan/khan-exercises/blob/master/interface.js#L484) succeeds and the user is informed that the answer is incorrect.